exchange/exchange.go

```diff
@@ -760,18 +754,18 @@ func (e *exchange) getAllBids(
 					}
 				}
 				// collect fledgeAuctionConfigs separately from bids, as empty seatBids may be discarded
-				fledge = collectFledgeFromSeatBid(fledge, bidderName, brw.adapter, seatBid)
+				auctionResponseExtraInfo.fledge = collectFledgeFromSeatBid(auctionResponseExtraInfo.fledge, bidderName, brw.adapter, seatBid)
 			}
 		}
 		//but we need to add all bidders data to adapterExtra to have metrics and other metadata
 		adapterExtra[brw.bidder] = brw.adapterExtra
 
-		if !bidsFound && adapterBids[brw.bidder] != nil && len(adapterBids[brw.bidder].Bids) > 0 {
-			bidsFound = true
+		if !auctionResponseExtraInfo.bidsFound && adapterBids[brw.bidder] != nil && len(adapterBids[brw.bidder].Bids) > 0 {
```

Comment:
Gus, `getAllBids` is called in if statement from `HoldAuction` because of StoredAuctionResponses. For "real" requests we always call `getAllBids`. 
This if statement is needed here to avoid overriding `bidsFound` value when auction request has more than one bidder. Imagine auction request to two bidders: Appnexus and Pubmatic. In `getAllBids` we run requests to bidders in parallel:
```go
for _, bidder := range bidderRequests {
   ...
   go bidderRunner(bidder, conversions)
   ...
}
``` 
And then read responses from channel. 
Let's say response from Appnexus came first and it had 1 bid back. We set `bidsFound` to true. 
Then response from Pubmatic came with no bids. Without this if statement Pubmatic will overwrite `bidsFound` to false, which is incorrect.

I believe this if statement is needed here for this reason. It looks complicated and confusing, but I don't see other ways to simplify it. 
Please correct me if I miss something.